finviz/helper_functions/request_functions.py

```diff
@@ -6,49 +6,72 @@
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
+CONCURRENT_CONNECTIONS = 20
+CONNECTION_TIMEOUT = 3000
+PROXY_URL = 'http://127.0.0.1:8080'
+PROXIES = {
+    'http': PROXY_URL,
+    'https': PROXY_URL
+}
 
 def http_request_get(url, session=None, payload=None, parse=True):
     """ Sends a GET HTTP request to a website and returns its HTML content and full url address. """
 
     if payload is None:
         payload = {}
 
-    if session:
-        content = session.get(url, params=payload, verify=False)
-    else:
-        content = requests.get(url, params=payload, verify=False)
+    try:
+        if session:
+            content = session.get(url, proxy=PROXY_URL, params=payload, verify_ssl=False)
+        else:
+            content = requests.get(url, proxies=PROXIES, params=payload, verify=False)
 
-    content.raise_for_status()  # Raise HTTPError for bad requests (4xx or 5xx)
-
-    if parse:
-        return html.fromstring(content.text), content.url
-    else:
-        return content.text, content.url
+        content.raise_for_status()  # Raise HTTPError for bad requests (4xx or 5xx)
 
+        if parse:
+            return html.fromstring(content.text), content.url
+        else:
+            return content.text, content.url
+    except (asyncio.TimeoutError, requests.exceptions.Timeout) as e:
+        print("Timed out while retrieving %s" % url)
+    except Exception as e:
+        print("Error retrieving %s (%s)" % (url, str(e)))
 
 class Connector(object):
     """ Used to make asynchronous HTTP requests. """
 
-    def __init__(self, scrape_function, tasks, *args):
+    def __init__(self, scrape_function, tasks, *args, cssselect=False):
 
         self.scrape_function = scrape_function
         self.tasks = tasks
         self.arguments = args
+        self.cssselect = cssselect
         self.data = []
 
     async def __http_request__async(self, url, session):
         """ Sends asynchronous http request to URL address and scrapes the webpage. """
 
-        async with session.get(url) as response:
-            page_html = await response.read()
+        try:
+            async with session.get(url, proxy=PROXY_URL, verify_ssl=False) as response:
+                page_html = await response.read()
 
-            return self.scrape_function(page_html, url=url, *self.arguments)
+                if self.cssselect is True:
+                    return self.scrape_function(html.fromstring(page_html), url=url, *self.arguments)
+                else:
+                    return self.scrape_function(page_html, url=url, *self.arguments)
+        except (asyncio.TimeoutError, requests.exceptions.Timeout) as e:
+            print("Timed out while retrieving %s" % url)
+        except Exception as e:
+            print("Error retrieving %s (%s)" % (url, str(e)))
```

Comment:
I feel that using print statements in this way is a bad practice. If you believe that `asyncio.TimeoutError` is ambiguous for the user, we can write a custom error class in `error_handling.py` and raise it from there. 

Finally, I don't think you can get `requests.exceptions.Timeout` error from this block of code, since it only uses `asyncio` and `aiohttp` for the requests. Correct me if I'm wrong.

Reply:
Not ideal to use `print()` indeed - are you going to work on `error_handling.py`?

Reply:
You're right, `requests.exceptions` shouldn't be hit in this block of code, it was a copy-paste from the other function where I did enclose in try/except block where both requests and session are used for the HTTP(S) request

Reply:
I can write the error handling today or tomorrow and submit and merge a new pull request. I will tag you whenever it's done.